tests/unit/test_choice_rule.py

```diff
@@ -61,6 +61,35 @@ def test_variable_value_must_be_consistent():
         with pytest.raises(ValueError):
             func('$.Variable', True)
 
+def test_path_function_value_must_be_consistent():
+    path_functions = {
+        'StringEqualsPath',
+        'NumericEqualsPath',
+        'TimestampEqualsPath',
+        'BooleanEqualsPath'
+    }
+    for path_function in path_functions:
+        func = getattr(ChoiceRule, path_function)
+        with pytest.raises(ValueError):
+            func('$.Variable', 'string')
+
+def test_is_function_value_must_be_consistent():
+    is_functions = {
+        'IsPresent',
+        'IsNull',
+        'IsString',
+        'IsNumeric',
+        'IsBoolean',
+        'IsTimestamp'
+    }
+
+    for is_function in is_functions:
+        func = getattr(ChoiceRule, is_function)
+        with pytest.raises(ValueError):
+            func('$.Variable', 'string')
+        with pytest.raises(ValueError):
+            func('$.Variable', 101)
+
 def test_rule_serialization():
```

Comment:
Can we add similar tests like this for the other choice rules to make sure they are serialized to the correct ASL? Since the rules that aren't boolean expressions more or less have the same pattern, it should be easy to iterate over them.

Reply:
added test for all choice rules serialization